arrow-fx-test/src/main/kotlin/arrow/test/laws/Law.kt

```diff
@@ -18,13 +18,13 @@ fun throwableEq() = Eq { a: Throwable, b ->
 data class Law(val name: String, val test: suspend TestContext.() -> Unit)
 
 fun <A> A.equalUnderTheLaw(b: A, eq: Eq<A>): Boolean =
-  eq.run { eqv(b) }
+  shouldBeEq(b, eq).let { true }
```

Comment:
Seems that this change unveiled some tests that weren't doing any verification with forAll or shouldBe and were failing 😓 cc @1Jajen1 @nomisRev 


arrow.fx.ResourceTest > Monad Laws: stack safe FAILED
arrow.fx.MVarTest > CancelableMVar - stack overflow test FAILED
arrow.fx.internal.TimeoutException: Duration(amount=5, timeUnit=SECONDS)
arrow.fx.MVarTest > CancelableMVar - producer-consumer parallel loop FAILED
arrow.fx.MVarTest > UncancelableMVar - concurrent take and put FAILED
arrow.fx.MVarTest > UncancelableMVar - stack overflow test FAILED
arrow.fx.MVarTest > UncancelableMVar - producer-consumer parallel loop FAILED
arrow.fx.IOTest > MonadDefer laws: stack safety over repeated left binds FAILED
arrow.fx.IOTest > MonadDefer laws: stack safety over repeated attempts FAILED
arrow.fx.SemaphoreTest > CancelableSemaphore - tryAcquire with available permits FAILED
arrow.fx.SemaphoreTest > UncancelableSemaphore - tryAcquire with available permits FAILED
arrow.fx.EqTest > Should fail for pure non-equal values FAILED